superbench/analyzer/data_diagnosis.py

```diff
@@ -396,29 +396,23 @@ def run(
             logger.info('DataDiagnosis: Begin to process {} nodes'.format(len(self._raw_data_df)))
             data_not_accept_df, label_df = self.run_diagnosis_rules(rules, baseline)
             logger.info('DataDiagnosis: Processed finished')
-            output_path = ''
+            output_path = str(Path(output_dir) / f'diagnosis_summary.{output_format}')
             # generate all nodes' info
             if output_all:
-                output_path = str(Path(output_dir) / 'diagnosis_summary.json')
                 data_not_accept_df = self.output_all_nodes_results(self._raw_data_df, data_not_accept_df)
```

Comment:
"not_accept_df = all_node_results" seems confusing, maybe rename to "output_df"?

Reply:
updated